plugin/v2/plugin.go

```diff
@@ -120,15 +125,37 @@ func (p *Plugin) InitPlugin(libRoot string) error {
 
 // Set is used to pass arguments to the plugin.
 func (p *Plugin) Set(args []string) error {
-	m := make(map[string]string, len(args))
-	for _, arg := range args {
-		i := strings.Index(arg, "=")
-		if i < 0 {
-			return fmt.Errorf("No equal sign '=' found in %s", arg)
+	p.Lock()
+	defer p.Unlock()
+
+	if p.PluginObj.Enabled {
+		return fmt.Errorf("cannot set on an active plugin, disable plugin before setting")
+	}
+
+	for _, s := range newSettables(args) {
+		// range over all the envs in the manifest
+		for _, env := range p.PluginObj.Manifest.Env {
+			// found the env in the manifest
+			if env.Name == s.name {
+				// is it settable ?
+				if !s.isSettable(allowedSettableFieldsEnv, env.Settable) {
+					return fmt.Errorf("%q is not settable", s.prettyName())
+				}
+				// is it, so lets update the config in memory
+				p.PluginObj.Config.Env = updateConfigEnv(p.PluginObj.Config.Env, &s)
+				goto next
```

Comment:
Not a fan of `goto next`. Can we set a flag if we found our env and it passed `isSettable` test and check it outside the loop?